src/reconcile/rec_row.c

```diff
@@ -831,7 +834,19 @@ __wt_rec_row_leaf(
              *
              * Repack the cell if we clear the transaction ids in the cell.
              */
-            if (vpack->raw == WT_CELL_VALUE_COPY) {
+            /* If requested, rewrite the overflow value. */
+            if (page->modify->compact_rewrite_ovfl && F_ISSET(vpack, WT_CELL_UNPACK_OVERFLOW) &&
```

Comment:
@keithbostic, can you please review the changes here? I have posted an error message from `test_compact03` that indicate that we are still holding on to the older cell somehow.